test/integration/client/client_test.go

```diff
@@ -1342,3 +1355,192 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
+	createData := etcd.GetEtcdStorageData()
+
+	// gather resources to test
+	_, resourceLists, err := client.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		t.Fatalf("Failed to get ServerGroupsAndResources with error: %+v", err)
+	}
+
+	for _, resourceList := range resourceLists {
+		for _, resource := range resourceList.APIResources {
+			if !strings.HasSuffix(resource.Name, "/status") {
+				continue
+			}
+			mapping, err := createMapping(resourceList.GroupVersion, resource)
+			if err != nil {
+				t.Fatal(err)
+			}
+			// both spec and status get wiped for CSRs,
+			// nothing is expected to be managed for it, skip it
+			if mapping.Resource.Resource == "certificatesigningrequests" {
+				continue
+			}
+
+			newResource, ok := createData[mapping.Resource]
+			if !ok {
+				t.Fatalf("no test data for %s.  Please add a test for your new type to etcd.GetEtcdStorageData().", mapping.Resource)
+			}
+			newObj := unstructured.Unstructured{}
+			if err := json.Unmarshal([]byte(newResource.Stub), &newObj.Object); err != nil {
+				t.Fatal(err)
+			}
+
+			namespace := ns.Name
+			if mapping.Scope == meta.RESTScopeRoot {
+				namespace = ""
+			}
+			name := newObj.GetName()
+
+			// etcd test stub data doesn't contain apiVersion/kind (!), but apply requires it
+			newObj.SetGroupVersionKind(mapping.GroupVersionKind)
+
+			rsc := dynamicClient.Resource(mapping.Resource).Namespace(namespace)
+			// apply to create
+			_, err = rsc.Apply(context.TODO(), name, &newObj, metav1.ApplyOptions{FieldManager: "create_test"})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if err := rsc.Delete(context.TODO(), name, *metav1.NewDeleteOptions(0)); err != nil {
+				t.Fatalf("deleting final object failed: %v", err)
+			}
+		}
+	}
+
+	hits1, misses1, entries1 := checkTransportMetrics(t, client)
+	// hit ratio
+	if (hits1*100)/(hits1+misses1) < 90 {
+		t.Fatalf("transport cache hit ratio %d lower than 90", (hits1*100)/(hits1+misses1))
+	}
+
+	aggregatorClient := aggregatorclient.NewForConfigOrDie(result.ClientConfig)
+	aggregatedAPI := &apiregistrationv1.APIService{
+		ObjectMeta: metav1.ObjectMeta{Name: "v1alpha1.wardle.example.com"},
+		Spec: apiregistrationv1.APIServiceSpec{
+			Service: &apiregistrationv1.ServiceReference{
+				Namespace: "kube-wardle",
+				Name:      "api",
+			},
+			Group:                "wardle.example.com",
+			Version:              "v1alpha1",
+			GroupPriorityMinimum: 200,
+			VersionPriority:      200,
+		},
+	}
+	_, err = aggregatorClient.ApiregistrationV1().APIServices().Create(context.Background(), aggregatedAPI, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 1000; i++ {
+		apiService, err := aggregatorClient.ApiregistrationV1().APIServices().Get(context.Background(), "v1alpha1.wardle.example.com", metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = aggregatorClient.ApiregistrationV1().APIServices().Update(context.Background(), apiService, metav1.UpdateOptions{})
```

Comment:
Can you mutate a label on this to make sure the update isn't short circuited?

Reply:
yeah, it now makes a difference, previously was not creating new transports, now

```
    client_test.go:1513: metric rest_client_transport_cache_entries 4
    client_test.go:1513: metric rest_client_transport_create_calls_total{result="hit"} 2063
    client_test.go:1513: metric rest_client_transport_create_calls_total{result="miss"} 4
```